Wrappers/Python/ccpi/optimisation/functions/Function.py

```diff
@@ -401,8 +436,11 @@ class TranslateFunction(Function):
     """
     
     def __init__(self, function, center):
-        
-        super(TranslateFunction, self).__init__(L = function.L) 
+        try:
+            L = function.L
+        except NotImplementedError as nie:
+            L = None
+        super(TranslateFunction, self).__init__(L = L) 
```

Comment:
We may well not set `L` in the init of `TranslateFunction` and call the property `L` of the `function` in it.